iotedgedev/modules.py

```diff
@@ -15,6 +16,47 @@ def __init__(self, envvars, utility, output, dock):
         self.dock = dock
         self.dock.init_registry()
 
+    def add(self, name, template):
+        self.output.header("CREATING MODULE")
+
+        cwd = self.envvars.MODULES_PATH
+        if name.startswith("_") or name.endswith("_"):
+            self.output.error("Module name cannot start or end with the symbol _")
+            return
+        elif not re.match("^[a-zA-Z0-9_]+$", name):
+            self.output.error("Module name can only contain alphanumeric characters and the symbol _")
+            return
+        elif os.path.exists(os.path.join(cwd, name)):
+            self.output.error("Module \"{0}\" already exists under {1}".format(name, os.path.abspath(self.envvars.MODULES_PATH)))
+            return
+
+        repo = "{0}/{1}".format(self.envvars.CONTAINER_REGISTRY_SERVER, name.lower())
+        if template == "csharp":
+            dotnet = DotNet(self.envvars, self.output, self.utility)
+            dotnet.install_module_template()
+            dotnet.create_custom_module(name, repo, cwd)
+        elif template == "nodejs":
+            cmd = "yo azure-iot-edge-module -n {0} -r {1}".format(name, repo)
```

Comment:
Thanks. Now it will check dependency and fail fast if the dependency is not installed. In addition, I also updated README to add instructions on setting up environment manually https://github.com/Azure/iotedgedev/pull/180/files#diff-04c6e90faac2675aa89e2176d2eec7d8